src/foam/parse/DateParser.js

```diff
@@ -0,0 +1,727 @@
+/**
+ * @license
+ * Copyright 2025 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.parse',
+  name: 'DateParser',
+
+  extends: 'foam.parse.AbstractParser',
+  
+  documentation: `
+    Comprehensive date and datetime parser that handles all formats from DateUtil.js.
+    Uses FOAM parser framework with alt() to support all date/datetime formats in a single parser.
+    Supports both date-only and datetime formats.
+
+    Usage:
+      var parser = foam.parse.DateParser.create();
+      var date = parser.parseString('2025-01-15');
+      var datetime = parser.parseString('2025-01-15T14:30:45');
+  `,
+
+  requires: [
+    'foam.parse.Grammar',
+    'foam.parse.Parsers'
+  ],
+
+  properties: [
+    {
+      name: 'baseGrammar_',
+      value: function(alt, anyChar, chars, optional, range, repeat, seq, str, sym) {
+        return {
+          START: sym('dateOrDatetime'),
+
+          // Main entry point - try all three main formats
+          dateOrDatetime: alt(
+            sym('yyyymmdd'),
+            sym('mmddyyyy'),
+            sym('yymmdd')
+          ),
+
+          // YYYYMMDD - tries all variants (compact with time, compact, separated)
+          // Covers: YYYYMMDDHHMMSS, YYYYMMDD, YYYY-MM-DD, YYYY/MM/DD with optional time
+          yyyymmdd: alt(
+            sym('yyyymmddhhmmss-compact'),
+            sym('yyyymmdd-compact'),
+            sym('yyyymmdd-sep')
+          ),
+
+          // Timezone format: Z or +/-HH:MM or +/-HHMM
+          timezone: alt(
+            'Z',
+            // +HH:MM format
+            seq(
+              chars('+-'),
+              repeat(range('0', '9'), null, 2),
```

Comment:
I can give you a 'max' parameter here if you like.

Reply:
would be nice, instead of the nulls